### ISSUE TITLE
Editor crashes on double-clicking target node when connecting a signal

### ISSUE THREAD

**Issue by StraToN** [godotengine/godot#20062]
**Godot version:**
Master, commit 4d9fde9

**OS/device including version:**
Mageia 6

**Issue description:**
Editor crashes when double clicking the target node to connect to a signal.
![image](https://user-images.githubusercontent.com/1289207/42448046-30a784e6-837c-11e8-9ad0-0678774a06cc.png)

**Steps to reproduce:**
Clone this project:
https://github.com/groud/CarGoingInCircles.git

Replace `level/tilemap.gd` script by the code below (defines a new signal "checkpoint_passed" and emit it in `_on_Checkpoint_checkpoint()` function):
```
extends StaticBody2D

signal lap_changed
signal checkpoint_passed

var lap = 0

var checkpoints = Array()
var nb_of_checkpoints = 3


func _on_Checkpoint_checkpoint(checkpoint_id):
	# Checkpoint logic
	if (checkpoint_id == 0):
		if (checkpoints == range(nb_of_checkpoints)):
			lap += 1
			emit_signal("lap_changed", lap)
		checkpoints = [0]
	else:
		if (checkpoints == range(checkpoint_id)):
			checkpoints = range(checkpoint_id+1)
			
	emit_signal("checkpoint_passed")
```

Save.

Open the `level/level.tscn`.
Select the `Level` node, the open the Node tab next to the Inspector tab (or click the wave icon on the `Level` node in the scene tree).
Double-click the `checkpoint_passed` signal or select it and hit "Connect". The connection panel opens.
Double click the Level (root) node in the list.
The editor crashes with backtrace: 

> handle_crash: Program crashed with signal 11
> Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
> [1] /lib64/libc.so.6(+0x338a0) [0x7fb1786e68a0] (??:0)
> [2] Vector<TreeItem::Cell>::size() const (/home/julian/Programming/godot/godot/master/./core/vector.h:115)
> [3] TreeItem::get_metadata(int) const (/home/julian/Programming/godot/godot/master/scene/gui/tree.cpp:304 (discriminator 2))
> [4] ConnectionsDock::_make_or_edit_connection() (/home/julian/Programming/godot/godot/master/editor/connections_dialog.cpp:439 (discriminator 2))
> [5] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/method_bind.gen.inc:56 (discriminator 4))
> [6] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/object.cpp:914 (discriminator 1))
> [7] Object::emit_signal(StringName const&, Variant const**, int) (/home/julian/Programming/godot/godot/master/core/object.cpp:1195 (discriminator 1))
> [8] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/julian/Programming/godot/godot/master/core/object.cpp:1241)
> [9] ConnectDialog::ok_pressed() (/home/julian/Programming/godot/godot/master/editor/connections_dialog.cpp:109 (discriminator 2))
> [10] AcceptDialog::_ok_pressed() (/home/julian/Programming/godot/godot/master/scene/gui/dialogs.cpp:387)
> [11] MethodBind0::call(Object*, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/method_bind.gen.inc:56 (discriminator 4))
> [12] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/object.cpp:914 (discriminator 1))
> [13] Object::emit_signal(StringName const&, Variant const**, int) (/home/julian/Programming/godot/godot/master/core/object.cpp:1195 (discriminator 1))
> [14] Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/julian/Programming/godot/godot/master/core/object.cpp:1241)
> [15] Tree::propagate_mouse_event(Point2i const&, int, int, bool, TreeItem*, int, Ref<InputEventWithModifiers> const&) (/home/julian/Programming/godot/godot/master/scene/gui/tree.cpp:1726 (discriminator 2))
> [16] Tree::_gui_input(Ref<InputEvent>) (/home/julian/Programming/godot/godot/master/scene/gui/tree.cpp:2603 (discriminator 5))
> [17] MethodBind1<Ref<InputEvent> >::call(Object*, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/method_bind.gen.inc:729 (discriminator 12))
> [18] Object::call_multilevel(StringName const&, Variant const**, int) (/home/julian/Programming/godot/godot/master/core/object.cpp:748 (discriminator 1))
> [19] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/julian/Programming/godot/godot/master/core/object.cpp:855)
> [20] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/home/julian/Programming/godot/godot/master/scene/main/viewport.cpp:1423 (discriminator 2))
> [21] Viewport::_gui_input_event(Ref<InputEvent>) (/home/julian/Programming/godot/godot/master/scene/main/viewport.cpp:1702 (discriminator 3))
> [22] Viewport::input(Ref<InputEvent> const&) (/home/julian/Programming/godot/godot/master/scene/main/viewport.cpp:2432 (discriminator 2))
> [23] Viewport::_vp_input(Ref<InputEvent> const&) (/home/julian/Programming/godot/godot/master/scene/main/viewport.cpp:1217)
> [24] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/method_bind.gen.inc:729 (discriminator 12))
> [25] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/julian/Programming/godot/godot/master/core/object.cpp:914 (discriminator 1))
> [26] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/julian/Programming/godot/godot/master/core/object.cpp:839)
> [27] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/julian/Programming/godot/godot/master/scene/main/scene_tree.cpp:251)
> [28] SceneTree::input_event(Ref<InputEvent> const&) (/home/julian/Programming/godot/godot/master/scene/main/scene_tree.cpp:407 (discriminator 6))
> [29] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/julian/Programming/godot/godot/master/main/input_default.cpp:392)
> [30] InputDefault::parse_input_event(Ref<InputEvent> const&) (/home/julian/Programming/godot/godot/master/main/input_default.cpp:260)
> [31] OS_X11::process_xevents() (/home/julian/Programming/godot/godot/master/platform/x11/os_x11.cpp:1915 (discriminator 3))
> [32] OS_X11::run() (/home/julian/Programming/godot/godot/master/platform/x11/os_x11.cpp:2624)
> [33] /home/julian/Programming/godot/godot/master/bin/godot.x11.tools.64(main+0xb5) [0xfdbb4b] (/home/julian/Programming/godot/godot/master/platform/x11/godot_x11.cpp:56)
> [34] /lib64/libc.so.6(__libc_start_main+0xf0) [0x7fb1786d3600] (??:0)
> [35] /home/julian/Programming/godot/godot/master/bin/godot.x11.tools.64(_start+0x29) [0xfdb9c9] (/home/iurt/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:121)
> -- END OF BACKTRACE --

**Comment by StraToN** [comment]
Oddly, I can't replicate this bug anymore :(
I'll leave it open nevertheless, since I still encounter it from time to time. Feel free to test it yourself though, or perhaps you encountered it in your own project.

**Comment by ibrahn** [comment]
looks like the same thing that caused #19001, #20131 should fix it

**Comment by akien-mga** [comment]
Also seems to be a duplicate of #19066 given the backtrace, and assuming fixed by #20131.
